### PR TITLE
Cherrypick very basic support for adding AARDFable

### PR DESCRIPTION
**Pull request by mbarnett**
Introduces support for extending individual instances of unowned classes with support for the ActsAsRDFable readback API. Completely preliminary and may need to change if it ends up being awkward in use.

Intended usage here is that, given an instance of ActiveStorage::Blob or similar, which has annotations in the db:

```ruby
blob = ... #... some code returning an ActiveStorage::Blob
ActsAsRDFable.add_annotation_bindings!(blob)

# blob now responds to AARDFable methods:

blob.rdf_annotations
```

(I've ripped out support for overriding table name as that's more up in the air and quasi-nonfunctional right now)

### CODE REVIEW

**Comment by mbarnett** [comment]
I'm don't think inheriting would work for this, in that the advantage of using ActiveStorage is that other gems and pieces of Rails are directly compatible with ActiveStorage. If we start using our own subclasses, then no other gem is going to be set up to work with JupiterStorage (or whatever) derived classes – just to give one example, the internals of the attachment mechanisms in Rails core [directly instantiate classes named ActiveStorage::Attachment](https://github.com/rails/rails/blob/a6f42938390138ae88d4d32a3a16ddcb1e61d3f2/activestorage/lib/active_storage/attached/changes/create_one.rb#L48) and directly switch on the ActiveStorage::Blob class identifier. None of that will work with a subclass without us rewriting the attachment code to use the new subclasses. 

Where we can, we want to hew as close to the standard Rails code as possible.

I definitely think this API could be really awkward if we're pulling up a list of storage attachments and then need to iterate through them all to bind them. 

Basically my philosophy of API design here is, let's see how awkward this is in practice, and that should give us a better picture of exactly what a more convenient API might be that lets us avoid monkeying around in Storage internals too much.

**Comment by lagoan** [comment]
I agree that this proposed method is the quickest way to move forward, I am just asking for long term maintainability. Someone coming in could look at the class definitions (or another high level area) and would know that our attachments and blobs can have predicates attached to them, instead of finding this in a single place right before it is used.

If we create our own subclasses Jupiter would work the same as it would still see it as ActiveStorage::*, the only difference is that the rails' back end would not have access to the rdf methods.

We would only need to tell our subclasses to continue using the same tables as ActiveStorage on the database, that way the ruby core works and Jupiter has the added functionality. On our side we are no longer working just with ActiveStorage::* but rdfable attachments and blobs, and at least for AIP, attachments and blobs will map to Filesets and Files.

And just to clarify, I see the benefit of specifying this relationship on Jupiter at a higher level and I am not married to the idea of subclassing.

**Comment by mbarnett** [comment]
> If we create our own subclasses Jupiter would work the same as it would still see it as ActiveStorage::*, the only difference is that the rails' back end would not have access to the rdf methods.

> We would only need to tell our subclasses to continue using the same tables as ActiveStorage on the database, that way the ruby core works and Jupiter has the added functionality. On our side we are no longer working just with ActiveStorage::* but rdfable attachments and blobs, and at least for AIP, attachments and blobs will map to Filesets and Files.

This _would_ be really nice, but it won't actually work that way as far as I can see?

I mean I want it to work this way, you're right in that it would solve a lot of uncleanliness, buuuut given some storage subclass:

```ruby
class Jupiter::Attachment < ActiveStorage::Attachment 
  self.table_name =  'active_storage_attachments'
  def rdf_methods
    # ...
  end
end
```

and some model:

```ruby
class Item
  has_many_attached :files
end
```

That `has_many_attached` [macro in ActiveStorage](https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/attached/model.rb#L112) runs a bunch of declarations in the context of Item that include (among other things):

```ruby
       def #{name}
            @active_storage_attached_#{name} ||= ActiveStorage::Attached::Many.new("#{name}", self)
          end


 has_many :"#{name}_attachments", -> { where(name: name) }, as: :record, class_name: "ActiveStorage::Attachment", inverse_of: :record, dependent: :destroy do
          def purge
            each(&:purge)
            reset
          end

          def purge_later
            each(&:purge_later)
            reset
          end
        end
        has_many :"#{name}_blobs", through: :"#{name}_attachments", class_name: "ActiveStorage::Blob", source: :blob
```
( ActiveStorage::Attached::Many delegates to the `_attachments` association if you chase it down, btw).

Note the hard-coded class names on the associations.

Even if we _save_ Jupiter::Attachments to the table, the net result is that on all of our readbacks, eg)

```ruby
item.files.each do |file| 
# ...do some stuff with file
end
```

`file` will be an instance of ActiveStorage::Attachment regardless of what it was when it was written,. simply because that's what all of the ActiveRecord internals have been told it to reify that table row as. We _can_ swizzle the class to downcast it to our subclass, but if we have to iterate through `item.files` to downclass each attachment that doesn't feel more or less awkward to me than having to iterate through it all to bind each instance to the convenience methods?

Like I said, this is all a bit up in the air right now as I feel this out, and I don't intend the API in this pull request to be anything close to final or anything more than the bare minimum to let you move forward with something that we can refactor later. 

To be honest I'm starting to think we might need to move away from the current in-model declaration anyway entirely in favor of something like a set of metadata presenters, just based on awkwardness I'm running into writing an OAI_DC renderer. That would end up solving this "Unowned class annotation" issue without any monkeypatching, since we'd always own the presentation wrapper.
